cmd/ical-notifier/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/alexflint/go-arg"
+	ics "github.com/arran4/golang-ical"
+	"github.com/jm-lemmi/ical-relay/datastore"
+	"github.com/jm-lemmi/ical-relay/helpers"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate ../../.github/scripts/generate-version.sh
+//go:embed VERSION
+var version string // If you are here due to a compile error, run go generate
+
+var configPath string
+var conf Config
+var dataStore datastore.DataStore
+
+func main() {
+	log.Info("Welcome to ical-notifier, version " + version)
+
+	// CLI Flags
+	var args struct {
+		Notifier     string `help:"Run notifier with given ID"`
+		Verbose      bool   `arg:"-v,--verbose" help:"verbosity level Debug"`
+		Superverbose bool   `arg:"--superverbose" help:"verbosity level Trace"`
+		ConfigPath   string `arg:"-c,--config" help:"Configuration path" default:"config.yml"`
+		// TODO: add data path
+	}
+	arg.MustParse(&args)
+
+	configPath = args.ConfigPath
+
+	if args.Verbose {
+		log.SetLevel(log.DebugLevel)
+	}
+	if args.Superverbose {
+		log.SetLevel(log.TraceLevel)
+	}
+
+	// load config
+	var err error
+	conf, err = ParseConfig(configPath)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	if !args.Verbose && !args.Superverbose {
+		// only set the level from config, if not set by flags
+		log.SetLevel(conf.General.LogLevel)
+	}
+
+	log.Debug("Debug log is enabled") // only shows if Debug is actually enabled
+	log.Trace("Trace log is enabled") // only shows if Trace is actually enabled
+
+	log.Tracef("%+v\n", conf)
+
+	if !helpers.DirectoryExists(conf.General.StoragePath + "notifystore/") {
+		log.Info("Creating notifystore directory")
+		err = os.MkdirAll(conf.General.StoragePath+"notifystore/", 0750)
+		if err != nil {
+			log.Fatalf("Error creating notifystore: %v", err)
+		}
+	}
+
+	// load data
+
+	if !conf.General.LiteMode {
+		// RUNNING FULL MODE
+		log.Debug("Running in full mode.")
+		if conf.General.DB.Host == "" {
+			log.Fatal("DB configuration missing")
+		}
+
+		// connect to DB
+		datastore.Connect(conf.General.DB.User, conf.General.DB.Password, conf.General.DB.Host, conf.General.DB.DbName)
+		dataStore = datastore.DatabaseDataStore{}
+
+	} else {
+		log.Warn("Running in lite mode. No connection with ical-relay assumed. No dynamic unsubscription possible.")
+		// TODO: add dynamic data path
+		dataStore, err = datastore.ParseDataFile(conf.General.StoragePath + "data.yml")
```

Comment:
why are we expecting the data file here and are using a CLI flag in ical-relay?

Reply:
how else whould ical-notifier in lite mode get its data?

Reply:
like ical-relay, by passing the argument `--import-data path/to/data.yml`

Reply:
import data ist ja in kombination mit der DB um daten zu importieren. das ist ja eine andere Funktion

Reply:
ach stimmt, mein Fehler! Dennoch bin ich eigentlich Fan davon Config Dateien so unabhängig wie möglich vom sonstigen Storage zu halten. Ist halt die Frage ob sich der "Aufwand" lohnt oder wir es doch so lassen

Reply:
Ja das sollte auf jeden fall noch eine eigene Flag bekommen, deswegen steht da das todo ja dran.

Reply:
Keine Ahnung, wie ich das übersehen konnte..